tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+        assert len(set(points)) == 10
+
+        # Create a new task:
+        second_task = self.Task(
+            max_trials=10,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        assert second_task._space.sample(10) == points
+    
+    def test_call_is_reproducible(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ Two tasks created with the same args, given the same point, should produce the same
+        results.
+        """
+        task_id = 0
+        first_task = self.Task(
+            max_trials=10,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+
+        point = first_task._space.sample(1, seed=first_task.seed)[0]
+
+        second_task = self.Task(
+            max_trials=10,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+
+        results = second_task(point)
+
+        assert len(results) == 1
+        assert results[0]["type"] == "objective"
+        first_objective = results[0]["value"]
+        second_results = second_task(point)
+        second_objective = second_results[0]["value"]
+
+        assert second_objective == first_objective
+
+    def test_call_twice_with_same_task_gives_same_result(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ When using the same task and different values for the same point, the results should
+        be identical.
+        """
+        task_id = 0
+        task = self.Task(
+            max_trials=10,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+
+        point = task._space.sample(1, seed=task.seed)[0]
+
+        first_results = task(point)
+        assert len(first_results) == 1
+        assert first_results[0]["type"] == "objective"
+        first_objective = first_results[0]["value"]
+
+        # Pass the same point under different forms and check that the results are exactly the same.
+        for second_point in [point, tuple(point), dict(zip(task._space.keys(), point))]:
+            second_results = task(second_point)
+            assert second_results[0]["name"] == first_results[0]["name"]
+            assert second_results[0]["type"] == first_results[0]["type"]
+            second_objective = second_results[0]["value"]
+
+            # TODO: Not sure why, but the two values are very close, but different!
```

Comment:
Is the point a numpy array and then converted to floats?

Reply:
Perhaps, not sure.

Reply:
Or is the meta-model running on a GPU?

Reply:
The meta-model is trained on the GPU yes, if cuda is available. I'm not sure I understand the original question though, would you mind clarifying?

Reply:
I was trying to understand why the values are very close but different. Them running on the GPU would explain why (slight difference in order of operations leading to slightly different results). Another reason would be the conversion from  numpy array to floats that could perhaps lead to some rounding issue but it should not be the case if numpy array is in float64.